inc/toolbox.class.php

```diff
@@ -941,111 +927,87 @@ static function commonCheckForUseGLPI() {
-                   "</td>";
-         $error = 2;
-
-      } else {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ok_min.png' alt=\"".
-               __s('The functionality is found - Perfect!'). "\" title=\"".
-               __s('The functionality is found - Perfect!')."\"></td>";
-      }
-      echo "</tr>";
-
-      // Test for mbstring extension.
-      echo "<tr class='tab_bg_1'><td class='left b'>".__('Mbstring extension test')."</td>";
-
-      if (!extension_loaded('mbstring')) {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ko_min.png'>".
-               __('Mbstring extension of your parser PHP is not installed')."></td>";
-         $error = 2;
-
-      } else {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ok_min.png' alt=\"".
-               __s('The functionality is found - Perfect!'). "\" title=\"".
-               __s('The functionality is found - Perfect!')."\"></td>";
-      }
-      echo "</tr>";
-
-      // Test for GD extension.
-      echo "<tr class='tab_bg_1'><td class='left b'>".__('GD extension test')."</td>";
-
-      if (!extension_loaded('gd')) {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/warning_min.png'>".
-                     __('GD extension of your parser PHP is not installed')."></td>";
-         $error = 1;
-
-      } else {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ok_min.png' alt=\"".
-                     __s('The functionality is found - Perfect!'). "\" title=\"".
-                     __s('The functionality is found - Perfect!')."\"></td>";
-      }
-      echo "</tr>";
-
-      // Test for zlib extension.
-      echo "<tr class='tab_bg_1'><td class='left b'>".__('Zlib extension test')."</td>";
-
-      if (!extension_loaded('zlib')) {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ko_min.png'>".
-                     __('Zlib extension of your parser PHP is not installed')."></td>";
-         $error = 2;
-
-      } else {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ok_min.png' alt=\"".
-                     __s('The functionality is found - Perfect!'). "\" title=\"".
-                     __s('The functionality is found - Perfect!')."\"></td>";
-      }
-      echo "</tr>";
-
-      // Test for curl extension.
-      echo "<tr class='tab_bg_1'><td class='left b'>".__('Curl extension test')."</td>";
-
-      if (!extension_loaded('curl')) {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ko_min.png'>".
-                     __('Curl extension of your parser PHP is not installed')."></td>";
-         $error = 2;
+      $needed_extentions = [
+         'mysqli'   => [
+            'title'     => __('MySQL Improved extension test'),
+            'success'   => __s('Ok - the MySQLi class exist - Perfect!'),
+            'error'     => __('You must install the MySQL Improved extension for PHP.')
```

Comment:
Indeed, we got a `$error = 1;` for the `gd` extension test; I did not see it. thank you!